mysql/util.go

```diff
@@ -76,7 +76,7 @@ var defaultLengthAndDecimalForCast = map[byte]lengthAndDecimal{
 	TypeString:     {0, -1}, // Flen & Decimal differs.
 	TypeDate:       {10, 0},
 	TypeDatetime:   {19, 0},
-	TypeNewDecimal: {11, 0},
+	TypeNewDecimal: {10, 0},
```

Comment:
Why change this? Is there any reference in the MySQL doc or source code?

Reply:
@XuHuaiyu this change is to solve https://github.com/pingcap/tidb/issues/23495, and please see https://github.com/pingcap/tidb/issues/23495#issuecomment-809512924